src/LaunchDarkly.OpenFeature.ServerProvider/Provider.StatusProvider.cs

```diff
@@ -0,0 +1,83 @@
+using System.Threading.Channels;
+using LaunchDarkly.Logging;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace LaunchDarkly.OpenFeature.ServerProvider
+{
+    public sealed partial class Provider
+    {
+        private class StatusProvider
+        {
+            private ProviderStatus _providerStatus = ProviderStatus.NotReady;
+            private object _statusLock = new object();
+            private Channel<object> _eventChannel;
+            private string _providerName;
+            private Logger _logger;
+
+            public StatusProvider(Channel<object> eventChannel, string providerName, Logger logger)
+            {
+                _eventChannel = eventChannel;
+                _providerName = providerName;
+                _logger = logger;
+            }
+
+            private void EmitProviderEvent(ProviderEventTypes type, string message)
+            {
+                var payload = new ProviderEventPayload
+                {
+                    Type = type,
+                    ProviderName = _providerName
+                };
+                if (message != null)
+                {
+                    payload.Message = message;
+                }
+
+                if (!_eventChannel.Writer.TryWrite(payload))
```

Comment:
Apparently this is bounded to 1, so I have to use the async version, which is going to make this chaotic.